sklearn/model_selection/_search.py

```diff
@@ -833,10 +844,15 @@ class GridSearchCV(BaseSearchCV):
             - A string, giving an expression as a function of n_jobs,
               as in '2*n_jobs'
 
-    iid : boolean, default=True
+    iid : boolean, default=None
```

Comment:
maybe should be `default='warn'`

Reply:
hm... that's not entirely clear, is it? though I don't really have a better idea. Usually we use None for deprecated parameters.

Reply:
Generally, I prefer describing the defaults than stating them here anyway, particularly when they are something semantically underspecified like "None". But seeing `default='warn'` or `default='deprecated'` in the signature is quite user-friendly, IMO. None is not.